wp-includes/class-wp-service-workers.php

```diff
@@ -51,10 +86,47 @@ class WP_Service_Workers extends WP_Scripts {
 	 */
 	public $output = '';
 
+	/**
+	 * Registered caching routes and scripts.
+	 *
+	 * @var array
+	 */
+	public $registered_caching_routes = array();
+
+	/**
+	 * Registered routes and files for precaching.
+	 *
+	 * @var array
+	 */
+	public $registered_precaching_routes = array();
+
 	/**
 	 * Initialize the class.
 	 */
 	public function init() {
+
+		if ( ! function_exists( 'list_files' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		$this->register(
+			'workbox-sw',
+			array( $this, 'get_workbox_script' ),
+			array()
+		);
+
+		$this->register(
+			'caching-utils-sw',
+			PWA_PLUGIN_URL . '/wp-includes/js/service-worker.js',
+			array( 'workbox-sw' )
+		);
+
+		if ( SCRIPT_DEBUG ) {
+			$this->register_cached_route( '/(wp-admin|wp-includes)/.*\.(?:js|css|gif|png|svg)/', self::STRATEGY_NETWORK_ONLY, array(), true );
```

Comment:
Come to think of it, should not this condition be removed entirely? If using `networkOnly` then this is just passing through to the network and it would be better to just skip caching the route entirely. In other words, this `if` could become:

```php
if ( ! SCRIPT_DEBUG ) {
    $this->precache_admin_assets();
}
```